### PR TITLE
Code coverage actions for Pull-requests and Push with veryGoodOpenSource Threshold Checking

### PR DESCRIPTION
**Pull request by Devesh21700Kumar**
1). This PR adds GitHub action for checking Test Coverage and uploading it on CodeQL
2). The initial threshold set is `15%`
3). Coverage Checked for both push and pull-requests
4). Added Codecov badge in Readme

**What kind of change does this PR introduce?**

bugfix

**Issue Number:**

Fixes #
#205 

**Snapshots/Videos:**
https://github.com/Devesh21700Kumar/talawa-admin/runs/4753092126?check_suite_focus=true
(Tested on my forked branch for push)

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

**Comment by Devesh21700Kumar** [comment]
> No coverage uploaded for pull request base

@yasharth291 @palisadoes 
Ple check this out

No coverage uploaded for pull request base (develop@66ecf14) as codecov has been used for the first time in this repo

File: .github/workflows/push.yml [hunk 1]
```diff
@@ -38,3 +38,21 @@ jobs:
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - run: yarn install
+      - run: yarn test --watchAll=false --coverage
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v1   
+      - name: Report code coverage
```
Comment (palisadoes) [inline .github/workflows/push.yml:54]: Remove everything from this line downwards. We don't want `VeryGoodOpenSource/very_good_coverage@v1.1.1` run again at this stage. Only `codecov/codecov-action@v1`

File: .github/workflows/README.md [hunk 2]
```diff
@@ -44,3 +44,7 @@ Follow these guidelines when creating or modifying scripts in this directory.
     1. Pylint
     1. Flake8
 1. All scripts must run a main() function.
+
+## Code Coverage
+
+[![codecov](https://codecov.io/gh/PalisadoesFoundation/talawa-api/branch/master/graph/badge.svg?token=CECBQTAOKM)](https://codecov.io/gh/PalisadoesFoundation/talawa-api)
```
Comment (palisadoes) [inline .github/workflows/README.md:50]: Change this to: 
```
[![codecov](https://codecov.io/gh/PalisadoesFoundation/talawa-admin/branch/develop/graph/badge.svg?token=II0R0RREES)](https://codecov.io/gh/PalisadoesFoundation/talawa-admin)
```

**Review by palisadoes — CHANGES_REQUESTED** [review]

File: .github/workflows/README.md (hunk 2 above)
Comment (Devesh21700Kumar) [reply .github/workflows/README.md:50]: Sure, will fix this now

File: .github/workflows/push.yml (hunk 1 above)
Comment (Devesh21700Kumar) [reply .github/workflows/push.yml:54]: Ok

File: .github/workflows/pull-requests.yml [hunk 3]
```diff
@@ -38,10 +38,28 @@ jobs:
...
+        with:
+          node-version: ${{ matrix.node-version }}
+      - run: yarn install
+      - run: yarn test --watchAll=false --coverage
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v1   
+      - name: Report code coverage
+        uses: VeryGoodOpenSource/very_good_coverage@v1.1.1
+        with:
+          path: "./coverage/lcov.info"
```
[the remark this answers is not in the document]
Comment (Devesh21700Kumar) [reply .github/workflows/pull-requests.yml:56]: > The `coverage/` directory needs to be added to the `.gitignore` file. Please add this.

image

Its already there in .gitignore

**Comment by Devesh21700Kumar** [comment]
@palisadoes, I made the changes

**Review by palisadoes — APPROVED** [review]

**Pull request merged**